Formula/xmrig.rb

```diff
@@ -23,16 +24,23 @@ class Xmrig < Formula
 
   depends_on "cmake" => :build
   depends_on "hwloc"
-  depends_on "libmicrohttpd"
   depends_on "libuv"
-  depends_on "openssl@1.1"
+  depends_on "openssl@3"
 
   def install
-    mkdir "build" do
-      system "cmake", "..", "-DWITH_CN_GPU=OFF", *std_cmake_args
-      system "make"
-      bin.install "xmrig"
-    end
+    # Use shared OpenSSL on macOS. In cmake/OpenSSL.cmake:
+    # elseif (APPLE)
+    #   set(OPENSSL_USE_STATIC_LIBS TRUE)
+    # endif()
+    inreplace "cmake/OpenSSL.cmake", "elseif (APPLE)", "elseif (OFF)"
```

Comment:
This `inreplace` looks a bit fragile. Can we handle `OPENSSL_USE_STATIC_LIBS` instead?